manifests/extract.pp

```diff
@@ -41,25 +41,34 @@
   $strip_components=0,
   $purge=false,
   $user=undef,
-  $tar_command=$::osfamily ? {
-    'Solaris' => 'gtar',
-    default   => 'tar',
-  },
+  $tar_command=undef,
 ) {
 
+  include ::archive::params
+
   if $root_dir {
     $extract_dir = "${target}/${root_dir}"
   } else {
     $extract_dir = "${target}/${name}"
   }
 
+  if ! $tar_command and ! $::archive::params::tarcmd {
+    fail("${module_name}: parameter \$tar_command not specified and \$::archive::params::tarcmd not found")
```

Comment:
If you include `archive::params` on line 47, there's no reason why you would get in a fail situation here. However, if you use `$::archive::params::tarcmd` as the default value for `$tar_command`, then you can check that `$tar_command` is indeed defined here and fail otherwise, and then you don't need the `$real_tar_command` variable below.